frontends/p4/structInitializers.cpp

```diff
@@ -13,10 +13,166 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-
 #include "structInitializers.h"
 
 namespace P4 {
+const IR::Expression*
+convert(const IR::Expression* expression, const IR::Type* type);
+
+const IR::Expression* defaultValue(const IR::Type* type, const Util::SourceInfo srcInfo) {
+    IR::Expression* expr;
+    if (type->is<IR::Type_Bits>()) {
+        expr = new IR::Constant(type, 0);
+    } else if (type->is<IR::Type_Boolean>()) {
+        expr = new IR::BoolLiteral(false);
+    } else if (type->is<IR::Type_Error>()) {
+        auto path = new IR::Path(IR::ID(srcInfo, "error"));
+        auto typeName =  new IR::Type_Name(srcInfo, path);
+        auto id = new IR::ID(srcInfo, "NoError");
+        auto tne = new IR::TypeNameExpression(srcInfo, typeName);
+        expr = new IR::Member(srcInfo, tne, *id);
+    } else if (auto t = type->to<IR::Type_Enum>()) {
+        BUG_CHECK(t->members.size(), "unknown default value for enum with no members");
+        auto tne = new IR::TypeNameExpression(srcInfo,
+                                               new IR::Type_Name(t->name));
+        expr = new IR::Member(srcInfo, tne, t->members.at(0)->name);
+    } else if (auto t = type->to<IR::Type_SerEnum>()) {
+        expr = new IR::Constant(t->type, 0);
+    } else if (type->is<IR::Type_StructLike>() || type->is<IR::Type_BaseList>()) {
+        auto vec = new IR::Vector<IR::Expression>();
+        auto conv = convert(new IR::ListExpression(srcInfo, *vec, true), type);
+        return conv;
+    } else {
+        ::error(ErrorType::ERR_TYPE_ERROR,
+                "%1%: Unknown default value for type %2%", srcInfo, type);
+    }
+    return expr;
```

Comment:
this may return an initialized variable expr; please initialize expr to nullptr.